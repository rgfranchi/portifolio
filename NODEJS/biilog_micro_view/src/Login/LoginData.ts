/**
 * Carrega informações de usuário.
 */
import { api } from "../config/api";
import { IDataValue } from "../interfaces/DataInterface";
import { errorRequest, verifyRequestValue } from "../helper/HttpRequestHelper";
import { ISessionSlice } from "../interfaces/SessionInterface";

export interface ILogin {
  email: string;
  password: string;
}

/**
 * Texto das entidades no singular e plural.
 */
export const displayName = { many: "Usuários", one: "Usuário" };

/**
 * Valores padrão vazio para preenchimento da entidade
 */
export const emptyLogin: ILogin = {
  email: "",
  password: "",
};

/**
 * Recupera valores de login do usuário.
 */
export const authenticationUser = async (
  email: string,
  password: string
): Promise<IDataValue<ISessionSlice>> => {
  let data: IDataValue<ISessionSlice> = { response: "", status: 0 };
  await api
    .get<ISessionSlice>(`user/authentication/${email}/${password}`)
    .then((resp) => {
      data = verifyRequestValue<ISessionSlice>(resp, displayName.one);
    })
    .catch((error) => {
      data = errorRequest(error, displayName.one);
      if (data.status === 401) {
        data.response = "Usuário ou Senha inválido";
      }
    });
  return data;
};
