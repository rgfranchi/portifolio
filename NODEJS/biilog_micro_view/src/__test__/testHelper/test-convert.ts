export const convertIdToNumeric = (data: any) => {
  if (Array.isArray(data)) {
    data.forEach((element) => {
      element = convertIdToNumeric(element);
    });
  }

  if (typeof data === "object" && data) {
    Object.entries(data).forEach((value, key) => {
      const k = value[0];
      const v = value[1];
      if (typeof v === "object") {
        data[k] = convertIdToNumeric(v);
      }
      if (k === "id") {
        const parse = parseInt(data[k]);
        if (!isNaN(parse)) {
          data[k] = parse;
        }
      }
    });
  }
  return data;
};

export const convertDateYYYYMMDDToDD_MM_YYYY = (dateSrt: string) => {
  const dArr = dateSrt.split("-"); // ex input "2010-01-18"
  return dArr[2] + "/" + dArr[1] + "/" + dArr[0]; //ex out: "18/01/2010"
};
