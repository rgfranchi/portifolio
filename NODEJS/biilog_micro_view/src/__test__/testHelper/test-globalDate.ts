/**
 * Fixa a data e hora do sistema para teste.
 * Realizar chamada no beforeAll(() => {});
 * Criar variÃ¡vel global : const realDate = Date.bind(global.Date);
 * E incluir no afterAll(() => {global.Date = realDate;});
 * @param YYYY
 * @param MM
 * @param DD
 * @param HH
 * @param mm
 * @param ss
 */

export const fixedGlobalDate = (
  YYYY: number,
  MM: number,
  DD: number,
  HH: number,
  mm: number,
  ss: number
) => {
  // https://github.com/facebook/jest/issues/2234#issuecomment-573621597
  const fixedDate = new Date(YYYY, MM, DD, HH, mm, ss);
  const d = Date;
  // This will only mock any Date objects instantiated with new
  // and not Date.now().
  const mockGlobal = global;
  // @ts-ignore
  mockGlobal.Date = jest.fn(() => fixedDate);
  mockGlobal.Date.parse = d.parse;
  mockGlobal.Date.UTC = d.UTC;
  mockGlobal.Date.now = d.now;
};
