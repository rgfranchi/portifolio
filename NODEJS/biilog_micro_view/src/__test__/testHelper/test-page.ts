import toJson from "enzyme-to-json";

/**
 * verifica o conteúdo de construção dos componentes Page.
 * @param {*} values -> array defaultPageText de test-text-defaults.js
 * @param {*} warpper -> componente que será testado, verifica Snapshot
 */
export const verifyPageContents = (values: any, wrapper: any) => {
  values.find.forEach((search: any) => {
    // console.log(search);
    expect(wrapper.find(search).exists()).toBeTruthy();
  });
  // console.log(toJson(wrapper));
  expect(toJson(wrapper)).toMatchSnapshot();
};

/**
 * verifica o conteúdo de construção dos componentes Page.
 * @param {*} values -> array defaultPageText de test-text-defaults.js
 * @param {*} warpper -> componente que será testado, verifica Snapshot
 */
export const verifyNotAccess = (values: any, wrapper: any) => {
  // console.log(values.find);
  values.find.forEach((search: any) => {
    // console.log(search);
    expect(wrapper.find(search).exists()).toBeFalsy();
  });
};

/**
 * verifica o conteúdo de construção dos componentes Page.
 * @param {*} values -> array defaultPageText de test-text-defaults.js
 * @param {*} history -> histórico da url para validação.
 */
export const loadHistoryPageContents = (values: any, history: any) => {
  history.entries[0].key = values.historyKey1;
  if (values.url1 !== "") {
    history.push(values.url1);
    history.entries[1].key = values.historyKey2;
  }
};
