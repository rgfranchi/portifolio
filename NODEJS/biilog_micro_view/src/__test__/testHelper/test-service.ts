import {
  screen,
  waitForElement,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { confirmModal } from "../../redux/modalSlice";

/**
 * Verifica se os elementos de save da tela foram executados com sucesso.
 * @param {boolean} isSuccess
 * @param {*} store
 */
export const verifyScreenAfterSave = async (isSuccess: boolean, store: any) => {
  // verifica se o modal Loading está ativo
  expect(screen.getByText(/Loading/i)).toBeInTheDocument();
  // verifica mensagem de falha.
  expect(store.getState().alert.hidden).toBe(true);
  // aguarda modal finalizar
  await waitForElementToBeRemoved(() => screen.getByText(/Loading/i));

  // Verifica estado dos elementos.
  if (isSuccess) {
    expect(store.getState().modals.success.show).toBe(true);
    expect(store.getState().modals.success.textButton).toMatch(/Retornar/i);
  } else {
    expect(store.getState().alert.hidden).toBe(false);
    expect(store.getState().alert.message).toMatch(/ERROR STATUS\[/i);
  }
};

/**
 * Executa comando de exclusão da tela List
 * @param {number} buttonPosition
 * @param {*} store
 */
export const verifyExecuteScreenDelete = async (
  buttonPosition: number,
  store: any
) => {
  // clica sobre o segundo link
  userEvent.click(screen.getAllByRole("button")[buttonPosition]);
  // verifica se alterou para exibir modal
  expect(store.getState().modals.confirm.show).toBe(true);
  // variável de NÃO executado.
  expect(store.getState().modals.confirm.executed).toBe(0);
  // realiza chamada de execução do modal.
  store.dispatch(confirmModal({ show: true, executed: 1 }));
  expect(store.getState().modals.confirm.executed).toBe(1);

  await waitForElement(() => screen.getByText(/Carregando/i));

  await waitForElementToBeRemoved(() => screen.getByText(/Carregando/i));
};

/**
 * Verifica e guarda carregar valores no tela, Create e Update
 * @param {boolean} isSuccess
 * @param {"modal" | "alert"} type "modal" | "alert"
 * @param {*} store
 */
export const loadingScreenValues = async (
  isSuccess: boolean,
  type: "modal" | "alert",
  store: any
) => {
  // state do loading exibido
  expect(store.getState().modals.loading.show).toBe(true);
  // elementos carregando Página de Loading
  expect(screen.getByText(/Carregando/i)).toBeInTheDocument();
  // verifica mensagem de falha.
  expect(store.getState().alert.hidden).toBe(true);
  // aguarda loading finalizar.
  await waitForElementToBeRemoved(() => screen.getByText(/Carregando/i));
  if (type === "modal") {
    if (isSuccess) {
      // state do loading oculto
      expect(store.getState().modals.loading.show).toBe(false);
    } else {
      expect(store.getState().modals.fail.show).toBe(true);
      expect(store.getState().modals.fail.textFail).toMatch(/vazia/i);
    }
  } else {
    expect(store.getState().alert.hidden).toBe(false);
    expect(store.getState().alert.message).toMatch(/vazia/i);
  }
};
