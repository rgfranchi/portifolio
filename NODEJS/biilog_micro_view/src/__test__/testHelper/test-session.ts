import { listGroupMock } from "../server/serverGroup";
import {
  ISessionAccess,
  ISessionSlice,
} from "../../interfaces/SessionInterface";

/**
 * Todo: Utilizar interface :ISessionSlice
 */
export const objMockEmptySession: ISessionSlice = {
  userId: 0,
  company: { id: 0, nome: "" },
  group: { id: 0, nome: "", access: [] },
};

export const objMockSession: ISessionSlice = {
  userId: 3,
  company: { id: 2, nome: "WINE SESSION" },
  group: listGroupMock[0],
};

export const objMockMenuAccess: ISessionAccess[] = [
  {
    path: "/company",
    page: {
      create: {
        label: "Nova Empresa",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
  {
    path: "/operator",
    page: {
      create: {
        label: "Inserir Operador",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
  {
    path: "/user",
    page: {
      create: {
        label: "Novo Usuário",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
  {
    path: "/machine",
    page: {
      create: {
        label: "Nova Máquina",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
  {
    path: "/forklift",
    page: {
      create: {
        label: "Inserir Equipamento",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
];
