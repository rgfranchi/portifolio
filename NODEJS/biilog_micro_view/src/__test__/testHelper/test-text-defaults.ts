import { IAlertSlice, ICompanySlice } from "../../interfaces/ReduxInterface";

export let defaultMessageTextData = {
  vaziaResponse: /vazia/i,
  falhaAcessarServidor: /Falha ao acessar servidor/i,
  idNaoDefinido: /ID não definido/i,
  criadoValor: "VALOR CRIADO XPTO",
  criadoSucesso: /Criado com sucesso/i,
  atualizadoValor: "VALOR ATUALIZADO XPTO",
  atualizadoSucesso: /Atualizado com sucesso/i,
  excluidoSucesso: "DELETED",
  erro208: "RETORNA_ERRO_208", // resposta Valida porem não esperada.
  erro208Mensagem: /INVALIDO Status/i,
  erro400: "RETORNA_ERRO_400", // resposta Valida porem não esperada.
  erro400Mensagem: /ERROR STATUS/i,
};

export let defaultFormTextValidate = {
  preenchimento: /Preenchimento obrigatório/i,
  minimo3caracteres: /com no mínimo 3 caracteres/i,
  emailInvalido: /e-mail inválido/i,
  cepFormato: /formato de 99999-999/i,
  cnpjInvalido: /CNPJ inválido/i,
  selecioneOpcao: /Selecione Uma Opção/i,
};

export let defaultPageText = {
  list: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_LIST1",
    url1: "",
    historyKey2: "",
    find: ["TitleComponent", "ButtonCreate", "List"],
  },
  create: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_NEW1",
    url1: "//new",
    historyKey2: "KEY_NEW2",
    find: ["TitleComponent", "CreateForm"],
  },
  update: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_UPDATE1",
    url1: "//update/2",
    historyKey2: "KEY_UPDATE2",
    find: ["TitleComponent", "UpdateForm"],
  },
};

export let alertSliceDefault: IAlertSlice = {
  hidden: true,
  message: null,
  variant: "primary",
};

export let companySliceDefault: ICompanySlice = {
  id: 3,
  nome: "Company Slice Mock",
};

// @ts-ignore
export let modalSliceDefault: IConfirmModal = {
  loading: {
    show: false,
    message: null,
  },
  confirm: {
    show: false,
    executed: 0, // incrementa a cada nova tentativa.
    disabledButtons: false,
    textTitle: null,
    textBody: null,
    textAction: "Executar",
    errorExecute: null,
    variantAction: "primary",
  },
  fail: {
    show: false,
    redirect: null,
    textFail: "Falha ao Executar",
    textButton: "Falhou ... ",
  },
  success: {
    show: false,
    redirect: null,
    textSuccess: "Sucesso!!!",
    textButton: "OK",
  },
};
