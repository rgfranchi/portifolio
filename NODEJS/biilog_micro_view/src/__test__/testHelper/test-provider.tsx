import React from "react";
import { MemoryRouter, Route, Router } from "react-router-dom";

import { objMockSession } from "./test-session";

import { Provider } from "react-redux";

//componentes de tela.
import { ModalLoading } from "../../components/ModalsComponent";

/**
 * Renderiza valores para teste.
 * Utilizando o render do framework @testing-library/react
 * Componente ModalLoading.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @returns
 */
export const renderProviderRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  // @ts-ignore
  function Wrapper({ children }) {
    return providerRouter(children, history, store, modal);
  }
  // @ts-ignore
  return render(ui, { wrapper: Wrapper });
};

/**
 * ConstrÃ³i com base no renderProviderRouter incluindo MemoryRouter e Router.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @param {*} initialEntries
 * @param {*} path
 * @returns
 */
export const renderProviderMemoryRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  initialEntries: any,
  path: string,
  modal: boolean = true
) => {
  const wrapper = (
    <MemoryRouter initialEntries={[initialEntries]}>
      <Route path={path}>{ui}</Route>
    </MemoryRouter>
  );
  return renderProviderRouter(render, wrapper, history, store, modal);
};

/**
 * Cria as funcionalidades do Provider e Router
 * Utilizando o mount do framework Enzyme
 * @param {*} mount
 * @param {*} children
 * @param {*} history
 * @param {*} store
 * @param {boolean} modal
 * @returns
 */
export const mountProviderRouter = async (
  mount: any,
  children: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  return await mount(providerRouter(children, history, store, modal));
};

export const createSessionCurrentUser = (store: any) => {
  store.dispatch({
    type: "session/currentUser",
    payload: objMockSession,
  });
};

const providerRouter = (
  children: any,
  history: any,
  store: any,
  modal: boolean
) => {
  return (
    <Provider store={store}>
      <Router history={history}>
        {modal && <ModalLoading />}
        {children}
      </Router>
    </Provider>
  );
};
