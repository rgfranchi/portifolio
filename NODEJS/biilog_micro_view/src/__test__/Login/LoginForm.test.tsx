import React from "react";

import { act } from "react-dom/test-utils";

import { render, fireEvent, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { createMemoryHistory } from "history";

import store from "../../redux/store";

import { renderProviderRouter } from "../testHelper/test-provider";

import { defaultFormTextValidate } from "../testHelper/test-text-defaults";

import LoginForm from "../../Login/LoginForm";
import { emptyLogin } from "../../Login/LoginData";
import {
  loadLoginInvalidValues,
  loadLoginValidValues,
  userMock,
} from "../server/serverUser";

let wrapper: any = null;
let history: any = null;
const objectMock = userMock;
const objectEmpty = emptyLogin;
const submitMock = jest.fn();

beforeEach(() => {
  cleanup();
  jest.setTimeout(20000);
  history = createMemoryHistory();
});

describe("LoginForm snapshot", () => {
  test("empty", async () => {
    history.entries[0].key = "EMPTY_01";
    await setWrapperDefault(objectEmpty, submitMock);
    expect(wrapper.container).toMatchSnapshot();
  });
  test("with data", async () => {
    history.entries[0].key = "DATA_01";
    await setWrapperDefault(objectMock, submitMock);
    expect(wrapper.container).toMatchSnapshot();
  });
});

describe("LoginForm validate", () => {
  test("valid", async () => {
    await setWrapperDefault(objectEmpty, submitMock);
    await act(async () => {
      await loadLoginValidValues(userEvent, screen);
    });
    expect(
      screen.queryByText(defaultFormTextValidate.preenchimento)
    ).toBeNull();
    await act(async () => {
      fireEvent.click(screen.getByText("Login"));
    });
    expect(submitMock).toHaveBeenCalledTimes(1);
  });
  test("invalid", async () => {
    await setWrapperDefault(objectEmpty, submitMock);
    await act(async () => {
      fireEvent.click(screen.getByText("Login"));
    });
    expect(screen.queryAllByText(/informar um E-mail/i).length).toBe(1);
    expect(screen.queryAllByText(/informar uma senha/i).length).toBe(1);
    await act(async () => {
      await loadLoginInvalidValues(userEvent, screen);
    });
    expect(
      screen.queryAllByText(defaultFormTextValidate.emailInvalido).length
    ).toBe(1);
  });
});

const setWrapperDefault = async (object: any, handleSubmit: any = () => {}) => {
  await act(async () => {
    wrapper = await renderProviderRouter(
      render,
      <LoginForm initValues={object} submitForm={handleSubmit} />,
      history,
      store
    );
  });
};

// Enzime.configure({ adapter: new Adapter() });

// beforeEach(() => {
//   cleanup();
// });

// test("LoginForm snapshot empty", () => {
//   const startStateConfirm = store.getState().modals.confirm;
//   const submitMock = jest.fn();
//   const wrapper = render(
//     <LoginForm initValues={emptyData} submitForm={submitMock} />,
//     {
//       initialState: {},
//       store: store,
//     }
//   );
//   expect(wrapper).toMatchSnapshot();
// });

// test("LoginForm submit Validade", async () => {
//   const startStateConfirm = store.getState().modals.confirm;
//   const submitMock = jest.fn();
//   render(<LoginForm initValues={emptyData} submitForm={submitMock} />, {
//     initialState: {},
//     store: store,
//   });
//   await act(async () => {
//     await userEvent.type(
//       screen.getByPlaceholderText("E-mail cadastrado"),
//       "operador@teste.com",
//       {
//         delay: 100,
//       }
//     );
//     await userEvent.type(screen.getByPlaceholderText("Senha"), "123456", {
//       delay: 100,
//     });
//   });

//   expect(screen.queryByText("É Necessário informar um E-mail")).toBeNull();
//   expect(screen.queryByText("Endereço de e-mail inválido")).toBeNull();
//   expect(screen.queryByText("É Necessário informar uma senha")).toBeNull();

//   // submit form.
//   await act(async () => {
//     fireEvent.click(screen.getByText("Login"));
//   });
//   expect(submitMock).toHaveBeenCalledTimes(1);
//   expect(submitMock.mock.calls[0][0]).toEqual({
//     email: "operador@teste.com",
//     password: "123456",
//   });
// });

// test("LoginForm submit Invalidate", async () => {
//   const startStateConfirm = store.getState().modals.confirm;
//   const submitMock = jest.fn();
//   render(<LoginForm initValues={emptyData} submitForm={submitMock} />, {
//     initialState: {},
//     store: store,
//   });

//   // submit form.
//   await act(async () => {
//     fireEvent.click(screen.getByText("Login"));
//   });

//   expect(screen.queryByText("É Necessário informar um E-mail")).toBeTruthy();
//   expect(screen.queryByText("É Necessário informar uma senha")).toBeTruthy();
//   await act(async () => {
//     await userEvent.type(
//       screen.getByPlaceholderText("E-mail cadastrado"),
//       "operador$teste.com",
//       {
//         delay: 100,
//       }
//     );
//   });
//   // submit form.
//   await act(async () => {
//     fireEvent.click(screen.getByText("Login"));
//   });
//   expect(screen.queryByText("Endereço de e-mail inválido")).toBeTruthy();
//   expect(submitMock).toHaveBeenCalledTimes(0);
// });
