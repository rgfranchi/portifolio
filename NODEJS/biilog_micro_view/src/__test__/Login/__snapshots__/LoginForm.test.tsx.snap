// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot empty 1`] = `
<div>
  <div
    class="login-form"
  >
    <form
      class=""
    >
      <h2
        class="text-center"
      >
        BIILOG
      </h2>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="email"
          placeholder="E-mail cadastrado"
          type="email"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="password"
          placeholder="Senha"
          type="password"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <button
          class="btn btn-primary btn-block"
          type="submit"
        >
          Login
        </button>
      </div>
      <div
        class="fade alert alert-primary alert-dismissible show"
        hidden=""
        role="alert"
      >
        <button
          class="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
          <span
            class="sr-only"
          >
            Close alert
          </span>
        </button>
        <ul>
          <li>
            null
          </li>
        </ul>
      </div>
      <div
        class="clearfix"
      >
        <a
          href="/recoverPassword"
        >
          Recuperar Senha ?
        </a>
      </div>
    </form>
  </div>
</div>
`;

exports[`LoginForm snapshot with data 1`] = `
<div>
  <div
    class="login-form"
  >
    <form
      class=""
    >
      <h2
        class="text-center"
      >
        BIILOG
      </h2>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="email"
          placeholder="E-mail cadastrado"
          type="email"
          value="testemock@teste.com.br"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="password"
          placeholder="Senha"
          type="password"
          value="dTVlclBAc3N3b3Jk"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group"
      >
        <button
          class="btn btn-primary btn-block"
          type="submit"
        >
          Login
        </button>
      </div>
      <div
        class="fade alert alert-primary alert-dismissible show"
        hidden=""
        role="alert"
      >
        <button
          class="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
          <span
            class="sr-only"
          >
            Close alert
          </span>
        </button>
        <ul>
          <li>
            null
          </li>
        </ul>
      </div>
      <div
        class="clearfix"
      >
        <a
          href="/recoverPassword"
        >
          Recuperar Senha ?
        </a>
      </div>
    </form>
  </div>
</div>
`;
